controller/query.go

```diff
@@ -0,0 +1,190 @@
+package controller
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/fabric8-services/fabric8-wit/app"
+	"github.com/fabric8-services/fabric8-wit/application"
+	"github.com/fabric8-services/fabric8-wit/errors"
+	"github.com/fabric8-services/fabric8-wit/jsonapi"
+	"github.com/fabric8-services/fabric8-wit/login"
+	"github.com/fabric8-services/fabric8-wit/query"
+	"github.com/fabric8-services/fabric8-wit/rest"
+	"github.com/fabric8-services/fabric8-wit/search"
+	"github.com/goadesign/goa"
+	"github.com/prometheus/common/log"
+)
+
+// QueryController implements the query resource.
+type QueryController struct {
+	*goa.Controller
+	db     application.DB
+	config QueryControllerConfiguration
+}
+
+// QueryControllerConfiguration the configuration for the LabelController
+type QueryControllerConfiguration interface {
+	GetCacheControlQueries() string
+	GetCacheControlQuery() string
+}
+
+// NewQueryController creates a query controller.
+func NewQueryController(service *goa.Service, db application.DB, config QueryControllerConfiguration) *QueryController {
+	return &QueryController{
+		Controller: service.NewController("QueryController"),
+		db:         db,
+		config:     config,
+	}
+}
+
+// Create runs the create action.
+func (c *QueryController) Create(ctx *app.CreateQueryContext) error {
+	currentUserIdentityID, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+	return application.Transactional(c.db, func(appl application.Application) error {
+		_, err = appl.Spaces().Load(ctx, ctx.SpaceID)
```

Comment:
Shouldn't you check for conditions like 'if ctx.SpaceID == nil' before using it?

Reply:
I think that `ctx.SpaceID` is not a pointer but a normal member.